gsy_framework/serialization.py

```diff
@@ -0,0 +1,17 @@
+import pickle
+from typing import Dict
+from zlib import compress, decompress
+
+
+class CompressSerialization:
+    """Custom Serializer / Deserializer."""
+    @staticmethod
+    def encode_and_compress_dict(uncompressed_dict):
+        """Convert a dict into str by pickling and compressing."""
+        pickled_str = pickle.dumps(uncompressed_dict)
+        return compress(pickled_str)
+
+    @staticmethod
+    def decompress_and_decode_queued_strings(compressed_dict: bytes) -> Dict:
```

Comment:
Here we don't need to know that we'll only decode "queued strings": the class is quite flexible, so it can be re-used for any kind of compressed byte data. Also, we're decompressing/decoding bytes, not strings (which would actually be dicts). Maybe something simpler would still work? For example
```suggestion
    def decompress_and_decode(compressed_dict: bytes) -> Dict:
```
Not sure if this would be optimal. What do you think?